pkg/build/strategies/sti/postexecutorstep.go

```diff
@@ -482,3 +451,102 @@ func createCommandForExecutingRunScript(scriptsURL map[string]string, location s
 	}
 	return cmd
 }
+
+func downloadAndExtractFileFromContainer(docker dockerpkg.Docker, tar s2itar.Tar, sourcePath, destinationPath, containerID string) (api.FailureReason, error) {
+	glog.V(5).Infof("Downloading file %q", sourcePath)
+
+	fd, err := ioutil.TempFile(destinationPath, "s2i-runtime-artifact")
+	if err != nil {
+		res := utilstatus.NewFailureReason(
+			utilstatus.ReasonFSOperationFailed,
+			utilstatus.ReasonMessageFSOperationFailed,
+		)
+		return res, fmt.Errorf("could not create temporary file for runtime artifact: %v", err)
+	}
+	defer func() {
+		fd.Close()
+		os.Remove(fd.Name())
+	}()
+
+	if err := docker.DownloadFromContainer(sourcePath, fd, containerID); err != nil {
+		res := utilstatus.NewFailureReason(
+			utilstatus.ReasonGenericS2IBuildFailed,
+			utilstatus.ReasonMessageGenericS2iBuildFailed,
+		)
+		return res, fmt.Errorf("could not download file (%q -> %q) from container %s: %v", sourcePath, fd.Name(), containerID, err)
+	}
+
+	// after writing to the file descriptor we need to rewind pointer to the beginning of the file before next reading
+	if _, err := fd.Seek(0, os.SEEK_SET); err != nil {
+		res := utilstatus.NewFailureReason(
+			utilstatus.ReasonGenericS2IBuildFailed,
+			utilstatus.ReasonMessageGenericS2iBuildFailed,
+		)
+		return res, fmt.Errorf("could not seek to the beginning of the file %q: %v", fd.Name(), err)
+	}
+
+	if err := tar.ExtractTarStream(destinationPath, fd); err != nil {
+		res := utilstatus.NewFailureReason(
+			utilstatus.ReasonGenericS2IBuildFailed,
+			utilstatus.ReasonMessageGenericS2iBuildFailed,
+		)
+		return res, fmt.Errorf("could not extract runtime artifact %q into the directory %q: %v", sourcePath, destinationPath, err)
```

Comment:
s/runtime// since this function is not strictly being used to extract runtime artifacts anymore, it's used for arbitrary content being pulled from a container.